pkg/apis/devconsole/v1alpha1/gitsource_types.go

```diff
@@ -50,8 +50,41 @@ type GitSourceStatus struct {
 	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
 	// Important: Run "operator-sdk generate k8s" to regenerate code after modifying this file
 	// Add custom validation using kubebuilder tags: https://book.kubebuilder.io/beyond_basics/generating_crd.html
+
+	// State represents current state of the GitSource, can be either initializing or ready
+	State State `json:"state"`
+
+	// Connection holds information whether the last attempt to reach the git source was successful or not. Optional
+	Connection Connection `json:"state,omitempty"`
 }
 
+// State represents current state of the GitSource,
+type State string
+
+// Initializing represents a state of GitSource whose creation wasn't completed and is not ready to use
+var Initializing State = "initializing"
```

Comment:
Maybe `const` instead `var` to avoid accidental overrides?
```suggestion
const Initializing State = "initializing"
```